### PR TITLE
VP-5035: Refactor migrations to simplify queries

### PR DESCRIPTION
**Pull request by akak1977**

### CODE REVIEW

File: src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs [hunk 1]
```diff
@@ -222,8 +222,8 @@ FROM INFORMATION_SCHEMA.TABLES
 										END
 								END,
 	                            null, null
-                            FROM [PlatformNotificationTemplate] pnt
-                            GROUP BY [NotificationTypeId]
+                            FROM [PlatformNotificationTemplate] pnt
+                            WHERE Language='en-US'
```
Comment (kostyrin) [inline src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs:226]: remove

File: src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs [hunk 2]
```diff
@@ -260,6 +260,8 @@ FROM INFORMATION_SCHEMA.TABLES
 											END
 									END
                               FROM [PlatformNotificationTemplate] pnt
+							  left outer join [Notification] n
```
Comment (kostyrin) [inline src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs:263]: all command chars need to be Upper case or use CAPS LOCK

File: src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs [hunk 3]
```diff
@@ -283,35 +285,37 @@ FROM INFORMATION_SCHEMA.TABLES
...
                                        ,[Message],
                                         [Discriminator])
-                            SELECT [Id]
-	                              ,[CreatedDate]
-                                  ,[ModifiedDate]
-                                  ,[CreatedBy]
-                                  ,[ModifiedBy]	
-	                              ,[ObjectId]
-                                  ,[ObjectTypeId]
-                                  , (SELECT TOP 1 n.Id FROM [Notification] n WHERE n.[Type] = pn.[Type])
```
Comment (kostyrin) [inline src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs:293]: also remove 'SELECT TOP 1..' on line 293

File: src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs [hunk 4]
```diff
@@ -283,35 +285,37 @@ FROM INFORMATION_SCHEMA.TABLES
...
+    										THEN 'EmailNotificationMessageEntity' 
+    										ELSE 
+    											CASE 
+    												WHEN pn.[Type] LIKE '%SmsNotification%'
+    												THEN 'SmsNotificationMessageEntity'
+    												ELSE 'EmailNotificationMessageEntity'
+    											END
+    									END
+                                  FROM [PlatformNotification] pn
+								  left outer join [Notification] n
```
Comment (kostyrin) [inline src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs:317]: UPPER CASE

**Review by kostyrin — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs [hunk 5]
```diff
@@ -181,49 +181,55 @@ FROM INFORMATION_SCHEMA.TABLES
...
+				                            THEN 'SmsNotificationEntity'
+				                            ELSE 'EmailNotificationEntity'
+			                            END
+	                            END,
+	                            null, null
+                            FROM 
+                            [PlatformNotificationTemplate] pnt
+                            INNER JOIN 
+                            (
+	                            SELECT [NotificationTypeId], MAX([Id]) [Id] FROM [PlatformNotificationTemplate]
```
Comment (yecli) [inline src/VirtoCommerce.NotificationsModule.Data/Migrations/20190528144557_InitialNotifications.cs:229]: Why are you using MAX, as it was TOP 1 before? Seems MIN is more suitable

**Pull request merged**